pkg/controller/volume/persistentvolume/index.go

```diff
@@ -157,6 +170,26 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *v1.PersistentVol
 	return nil, nil
 }
 
+// checkVolumeModeMatches is a convenience method that checks volumeMode for PersistentVolume
+// and PersistentVolumeClaims along with making sure that the Alpha feature gate BlockVolume is
+// enabled.
+// This is Alpha and could change in the future.
+func checkVolumeModeMatches(pvcSpec v1.PersistentVolumeClaimSpec, pvSpec v1.PersistentVolumeSpec) (bool, bool) {
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		requestedVolumeMode := pvcSpec.VolumeMode
+		pvVolumeMode := pvSpec.VolumeMode
+
+		if requestedVolumeMode != pvVolumeMode {
+			return true, false
+		}
+		return true, true
+	} else {
+		// feature gate is disabled
+		return false, false
```

Comment:
You could just return matches = true if the feature is disabled.  Then you don't need to return a feature enabled bool.

Reply:
yeah, I switched that up as well